platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteInput.java

```diff
@@ -231,12 +231,18 @@ protected void onBlur() {
 		onChange();
 	}
 
-	protected void onBackdropClickOrEscape() {
+	protected void onBackdropClick() {
 
-		closePopup();
-		deduceValue();
-		onChange();
-		inputField.focus();
+		closePopupAndDeduceValueAndHandleChange();
+	}
+
+	protected void onEscape() {
+
+		boolean popupWasAppended = popup.isAppended();
+		closePopupAndDeduceValueAndHandleChange();
+		if (!popupWasAppended) {
+			sendEscKeyToParent();
+		}
```

Comment:
I wonder if this would be easier to grasp?
```suggestion
		if (popup.isAppended()) {
			closePopupAndDeduceValueAndHandleChange();
		} else {
			closePopupAndDeduceValueAndHandleChange();
			sendEscKeyToParent();
		}
```
Or we replace `closePopupAndDeduceValueAndHandleChange` with `deduceValueAndHandleChange`:
```suggestion
		deduceValueAndHandleChange();

		if (popup.isAppended()) {
			closePopup();
		} else {
			sendEscKeyToParent();
		}
```
Hmm, on the other hand, do we really need to `deduceValueAndHandleChange()` if the user pressed ESC?

Reply:
Opted for the former.